src/Microsoft.AspNetCore.Authentication.OpenIdConnect/OpenIdConnectHandler.cs

```diff
@@ -110,12 +110,12 @@ public override Task<bool> HandleRequestAsync()
             var remoteSignOutContext = new RemoteSignOutContext(Context, Scheme, Options, message);
             await Events.RemoteSignOut(remoteSignOutContext);
 
-            if (remoteSignOutContext.HandledResponse)
+            if (remoteSignOutContext.State == EventResultState.HandleResponse)
             {
                 Logger.RemoteSignOutHandledResponse();
                 return true;
             }
-            if (remoteSignOutContext.Skipped)
+            if (remoteSignOutContext.State == EventResultState.SkipToNextMiddleware)
```

Comment:
> Longer term I think we should draw the line at some point for events, with subclass + overriding virtual methods being the full control route.

It's the pattern you use in Identity and it's often a PITA compared to the events model as the methods you expose are often useless (because their signatures are incomplete or because they are internal)... even simple things like flowing a claim from an external identity to the application cookie are painful to implement and require rewriting tons of code (try it!).